pkg/resources/server_test.go

```diff
@@ -0,0 +1,327 @@
+			err := rs.Start()
+			Expect(err).ToNot(HaveOccurred())
+		})
+		It("start server with plugin watcher disabled", func() {
+			grpcServer := grpc.NewServer([]grpc.ServerOption{}...)
+			rsc := &mocks.ResourceServerConnector{}
+			rsc.On("GetServer", mock.Anything).Return(grpcServer)
+			rsc.On("CreateServer").Return()
+			rsc.On("Listen", mock.Anything, mock.Anything).Return(nil, nil)
+			rsc.On("Serve", mock.Anything).Return()
+			rsc.On("Dial", mock.Anything, mock.Anything).Return(nil, nil)
+			rsc.On("Close", mock.Anything).Return()
+			rsc.On("Register", mock.Anything, mock.Anything).Return(nil)
+			rs := resourceServer{watchMode: false, rsConnector: rsc}
+			err := rs.Start()
+			Expect(err).ToNot(HaveOccurred())
+		})
+		It("start server with failing to listen to socket", func() {
+			rsc := &mocks.ResourceServerConnector{}
+			rsc.On("CreateServer").Return()
+			rsc.On("Listen", mock.Anything, mock.Anything).Return(nil, errors.New("failed"))
+			rs := resourceServer{rsConnector: rsc}
+			err := rs.Start()
+			Expect(err).To(HaveOccurred())
+		})
+		It("start server with plugin failed to dial server", func() {
+			grpcServer := grpc.NewServer([]grpc.ServerOption{}...)
+			rsc := &mocks.ResourceServerConnector{}
+			rsc.On("GetServer", mock.Anything).Return(grpcServer)
+			rsc.On("CreateServer").Return()
+			rsc.On("Listen", mock.Anything, mock.Anything).Return(nil, nil)
+			rsc.On("Serve", mock.Anything).Return()
+			rsc.On("Dial", mock.Anything, mock.Anything).Return(nil, errors.New("failed"))
+			rs := resourceServer{rsConnector: rsc}
+			err := rs.Start()
+			Expect(err).To(HaveOccurred())
+		})
+		It("start server with plugin watcher disabled failed to register", func() {
+			grpcServer := grpc.NewServer([]grpc.ServerOption{}...)
+			rsc := &mocks.ResourceServerConnector{}
+			rsc.On("GetServer", mock.Anything).Return(grpcServer)
+			rsc.On("CreateServer").Return()
+			rsc.On("Listen", mock.Anything, mock.Anything).Return(nil, nil)
+			rsc.On("Serve", mock.Anything).Return()
+			rsc.On("Dial", mock.Anything, mock.Anything).Return(nil, nil)
+			rsc.On("Close", mock.Anything).Return()
+			rsc.On("Stop").Return()
+			rsc.On("Register", mock.Anything, mock.Anything).Return(errors.New("failed"))
+			rs := resourceServer{watchMode: false, rsConnector: rsc}
+			err := rs.Start()
+			Expect(err).To(HaveOccurred())
+		})
+	})
+	Context("Stop", func() {
+		It("stop server with correct parameters", func() {
+			grpcServer := grpc.NewServer([]grpc.ServerOption{}...)
+			rsc := &mocks.ResourceServerConnector{}
+			rsc.On("DeleteServer").Return()
+			rsc.On("GetServer").Return(grpcServer)
+			rsc.On("Stop").Return()
+
+			stopWatcher := make(chan bool)
+			rs := resourceServer{
+				rsConnector: rsc,
+				stopWatcher: stopWatcher,
+				stop:        make(chan interface{}),
+			}
+			// Dummy listener to stopWatcher to not block the test and fail
+			go func() {
+				<-stopWatcher
+			}()
+
+			err := rs.Stop()
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
you should test that no events are in the channel then close it imo

Reply:
also add test when watchMode is true and ensure True is in the channel

Reply:
Split to 2 cased with/without watchMode

Reply:
where did you do that change?

Reply:
@moshe010 the cases are in line 175, but don't know why it doesn't show it in the code box above https://github.com/Mellanox/k8s-rdma-shared-dev-plugin/pull/3/files#diff-99dc642105db7e0063b2d11a583ea70eR175